### DISCUSSION TITLE
How to handle 'bad' data from the carriers?

### DISCUSSION THREAD

**Discussion by ChrisNolan** [karrioapi/karrio#650] [Q&A] [answered]
I have a bunch of tracking numbers from Fedex atm who are returning poorly formatted dates which Karrio is not able to handle atm.

In people's opinions, how should Karrio handle this do you think?

In this case for example Fedex is giving "date": "2024-08-15T00:00:00", when the code is expecting a timezone to be included -- e.g. 
`lib.to_date(event.date, "%Y-%m-%dT%H:%M:%S%z") for event in item.scanEvents`

I think it is a weird issue with a device in one of our warehouses that is causing it, but I'm not sure.  But it means a bunch of tracking numbers can't be added in the mean time, and before I 'hack' it up too much myself, I thought I'd ask people's opinions on 'best practices' for handling it?

Should I just add in a timezone if it is missing (and if so, go with "+0000" (or is it -0000), or try to 'infer' a timezone from neighbouring 'events' or something else in the 'header'.  Or should I more gracefully 'skip' that event, and if I skip the event, should I inject an event into the history saying that there was a missing event?  Or some combination of thsoe, or something else entirely?

Thanks for your time.  (Note I did email hello@karrio.io as well, sorry for the duplication)

**Comment by ChrisNolan** [comment]
[test1.json](https://github.com/user-attachments/files/16728843/test1.json)
Here is the full JSON from that example tracking number of 738488882438 which is one of many with the issue atm.

**Comment by danh91** [comment, marked as answer]
Addressed in https://github.com/karrioapi/karrio/issues/654
